atom/common/promise_util.cc

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2018 GitHub, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "atom/common/promise_util.h"
+
+#include <string>
+
+namespace atom {
+
+namespace util {
+
+Promise* Promise::RejectWithErrorMessage(const std::string& string) {
+  v8::Local<v8::String> error_message =
+      v8::String::NewFromUtf8(isolate(), string.c_str());
+  v8::Local<v8::Value> error = v8::Exception::Error(error_message);
+  GetInner()->Reject(mate::ConvertToV8(isolate(), error));
+  return this;
+}
+
+v8::Local<v8::Object> Promise::GetHandle() const {
+  return GetInner()->GetPromise();
+}
+
+}  // namespace util
+
+}  // namespace atom
+
+namespace mate {
+
+v8::Local<v8::Value> mate::Converter<atom::util::Promise*>::ToV8(
+    v8::Isolate* isolate,
```

Comment:
`isolate` should be renamed `unused`

Reply:
Again consistency, if unused is a standard we should follow I think we need to add to the style guide and rename a whole bunch of things

Reply:
Quite often an argument name is way more meaningful than it's type.
So `unused` might make code less clear.

From [Google C++ styleguide](https://google.github.io/styleguide/cppguide.html#Function_Declarations_and_Definitions):

> Unused parameters that might not be obvious should comment out the variable name in the function definition:
```cpp
class Shape {
 public:
  virtual void Rotate(double radians) = 0;
};

class Circle : public Shape {
 public:
  void Rotate(double radians) override;
};

void Circle::Rotate(double /*radians*/) {}
```
```cpp
// Bad - if someone wants to implement later, it's not clear what the
// variable means.
void Circle::Rotate(double) {}
```

Reply:
That appears to be about omitting the variable name altogether, which is a common C++ idiom for avoiding a -Wunused-param warning. This is also the case for [v8](https://bugs.chromium.org/p/v8/issues/detail?id=2302); omitting the variable name either by removing it or commenting out will hit the pre-commit scripts.

However in this case, I'm not sure that renaming "v8::Isolate isolate" to another name would make it any less clear, as the name is literally the data type. Also we have a long track record of using it in this way:

```sh
$ cd atom
$ ag unused,
browser/api/atom_api_box_layout.cc:76:                v8::Local<v8::Value> unused,
browser/api/atom_api_layout_manager.cc:52:                v8::Local<v8::Value> unused,
browser/api/atom_api_debugger.cc:170:                v8::Local<v8::Value> unused,
browser/api/atom_api_web_contents.cc:2142:                v8::Local<v8::Value> unused,
browser/api/atom_api_content_tracing.cc:62:                v8::Local<v8::Value> unused,
browser/api/atom_api_view.cc:75:                v8::Local<v8::Value> unused,
browser/api/atom_api_auto_updater.cc:144:                v8::Local<v8::Value> unused,
browser/api/atom_api_screen.cc:161:                v8::Local<v8::Value> unused,
browser/api/atom_api_in_app_purchase.cc:128:                v8::Local<v8::Value> unused,
browser/api/atom_api_app.cc:1338:                v8::Local<v8::Value> unused,
browser/api/atom_api_session.cc:863:                v8::Local<v8::Value> unused,
browser/api/atom_api_tray.cc:250:                v8::Local<v8::Value> unused,
browser/api/atom_api_global_shortcut.cc:93:                v8::Local<v8::Value> unused,
browser/api/atom_api_browser_view.cc:146:                v8::Local<v8::Value> unused,
browser/api/atom_api_desktop_capturer.cc:161:                v8::Local<v8::Value> unused,
browser/api/atom_api_notification.cc:256:                v8::Local<v8::Value> unused,
browser/api/atom_api_menu.cc:214:                v8::Local<v8::Value> unused,
browser/api/atom_api_dialog.cc:126:                v8::Local<v8::Value> unused,
browser/api/atom_api_system_preferences.cc:95:                v8::Local<v8::Value> unused,
browser/api/atom_api_top_level_window.cc:1073:                v8::Local<v8::Value> unused,
browser/api/atom_api_web_view_manager.cc:47:                v8::Local<v8::Value> unused,
browser/api/atom_api_power_monitor.cc:138:                v8::Local<v8::Value> unused,
browser/api/atom_api_render_process_preferences.cc:80:                v8::Local<v8::Value> unused,
browser/api/atom_api_power_save_blocker.cc:127:                v8::Local<v8::Value> unused,
browser/api/atom_api_web_contents_view.cc:117:                v8::Local<v8::Value> unused,
browser/api/atom_api_protocol.cc:237:                v8::Local<v8::Value> unused,
browser/api/atom_api_download_item.cc:229:                v8::Local<v8::Value> unused,
browser/api/atom_api_net.cc:47:                v8::Local<v8::Value> unused,
browser/api/atom_api_browser_window.cc:428:                v8::Local<v8::Value> unused,
common/api/atom_api_clipboard.cc:199:                v8::Local<v8::Value> unused,
common/api/features.cc:27:                v8::Local<v8::Value> unused,
common/api/atom_api_v8_util.cc:107:                v8::Local<v8::Value> unused,
common/api/atom_api_crash_reporter.cc:48:                v8::Local<v8::Value> unused,
common/api/atom_api_asar.cc:145:                v8::Local<v8::Value> unused,
common/api/atom_api_native_image.cc:610:                v8::Local<v8::Value> unused,
common/api/atom_api_shell.cc:140:                v8::Local<v8::Value> unused,
renderer/api/atom_api_renderer_ipc.h:24:                v8::Local<v8::Value> unused,
renderer/api/atom_api_renderer_ipc.cc:63:                v8::Local<v8::Value> unused,
renderer/api/atom_api_web_frame.cc:509:                v8::Local<v8::Value> unused,
```